packages/toolbars-v3/web/src/resolvers/tiptapResolvers.ts

```
@@ -352,75 +353,85 @@ export const getLineSpacingAfterSelectionResolver = TiptapContentResolver.create
+  (content, { styles, nodeService }: { styles: RicosStyles; nodeService }, editor): string => {
+    const getDefaultParagraphValue = () => {
+      const decoration = styles.getDecoration(
+        {
+          paragraphData: {},
+          type: Node_Type.PARAGRAPH,
+        } as RichTextNode,
+        Decoration_Type.FONT_SIZE
+      );
+      return decoration.fontSizeData?.value?.toString() ?? UNKNWON_FONT_SIZE;
+    };
 
-          currentFontSize = currentFontSize || fontSizeInDocumentStyle;
+    if (!editor) {
+      // @TODO: fix editor type for the resolvers
+      return getDefaultParagraphValue();
+    }
+    const { state } = editor;
 
-          if (returnedFontSize !== '' && returnedFontSize !== currentFontSize) return '';
-          returnedFontSize = currentFontSize;
-        }
-        const storedMark = editor?.state?.storedMarks?.find(mark => {
-          return mark.type.name === Decoration_Type.FONT_SIZE;
-        });
+    const { from, $from, to, empty } = state.selection;
+    if (empty) {
+      // From stored marks
+      const storedFontSize = state.storedMarks?.find(
+        m => m.type.name === Decoration_Type.FONT_SIZE
+      );
+      if (storedFontSize) {
+        return storedFontSize.attrs.value ?? 0;
+      }
+      // From mark under cursor
+      const markUnderCursor = $from.marks().find(m => m.type.name === Decoration_Type.FONT_SIZE);
+      if (markUnderCursor) {
+        return markUnderCursor.attrs.value ?? 0;
+      }
+      // From parent block node
+      const parent = state.doc.resolve(from).parent;
+      const ricosNode = fromTiptapNode({ ...parent.toJSON(), type: parent?.type?.name });
+      const decoration = styles.getDecoration(ricosNode as RichTextNode, Decoration_Type.FONT_SIZE);
+
+      if (decoration.fontSizeData?.value) {
+        const fontSizeInDocumentStyle = decoration.fontSizeData?.value;
+        return `${fontSizeInDocumentStyle}`;
+      }
+      // @TODO: should we throw an error here?
```

What would that mean? Would we break the editor? Not sure if this would good solution, we should try recovering/fallbacking as much as possible

---

Failing fast makes app better, but I'm not sure if its already a time for it